### DISCUSSION TITLE
Autocomplete and immediate feedback doesn't work

### DISCUSSION THREAD

**Discussion by cehan-Chloe** [0no-co/gql.tada#81] [Q&A] [answered]
Hi there!
This is awesome. I really like gql.tada and thinking about adding gql.tada to our monorepo. I'm following this [instruction](https://gql-tada.0no.co/) now. Here are some quick wins I got which is exciting but also some problems I'm facing and hope can get your help. Thanks in advance!

Status quo: 
1. Important dependencies: `apollo` v2, `graphql` 14.7.0, `react` ^18.2.0, `typescript` 5.3.3
2. gql.tada progress: 
2.1 graphql-env.d.ts files are generated. 
2.2 imported `graphql` from `gql.tada` and replaced `gql` from `graphql-tag` with it. tested end to end. No issue. 

Question:
1. some functions not working. Could you help to give some hints that I can continue debugging?
1.1 Not seeing hints when adding new fields to a query. eg. in the [demo](https://gql-tada.0no.co/), when adding new fields in `PokemanQuery`
1.2 When a field's usage is removed, hint is not shown in query that this field is no longer in use. referring another [demo](https://twitter.com/mattpocockuk/status/1760240498604642324?s=46&t=xBdZAfU1GpqWlZImSrX5hQ) 
2. What if schema is changed, how to regenerate the `graphql-env.d.ts` type files? 

```
// import
import { graphql, type ResultOf } from 'gql.tada';

// query
export const GET_BUSINESS_INFO = graphql(`
    query FederationBeta_GetBusinessInfo($id: String!) {
        business(id: $id) {
            name
            ...
        }
    }
`);

// useQuery
const { loading, error, data } = useQuery<
ResultOf<typeof GET_BUSINESS_INFO>>(GET_BUSINESS_INFO, {    
variables: { id },
    });

// component
<Code>
    <Pre> test</Pre>
    <Pre> {data?.business?.name}</Pre>
</Code> 
```

**Comment by kitten** [comment, marked as answer]
Sorry, didn't spot this back then.
We now have a `gql.tada doctor` command which may be able to detect problems in your setup.

**Reply by cehan-Chloe** [reply]
thanks for answering the question. When I run `gql.tada doctor` following the doc [here](https://gql-tada.0no.co/reference/gql-tada-cli#doctor), I get output `bash: gql.tada: command not found`

I upgraded the dependencies to "gql.tada": "^1.6.3", "@0no-co/graphqlsp": "^1.12.1". Still no luck

**Reply by kitten** [reply]
Depending on which package manager you're using, how your `PATH` is set up, and how the binaries are resolved, you may have to either update your environment for npm-installed binaries in general, or run them via the package manager (i.e. `npx gql.tada`, `pnpm gql.tada`, `bunx gql.tada`, `yarn gql.tada`, etc etc)

**Reply by cehan-Chloe** [reply]
`yarn gql.tada doctor` works for me! thank you so much

Now i got  
✘ Checking schema

 ⚠ Unexpected Error 
```
Failed to load schema.
┗ TypeError: n.executeSync is not a function
      at load (/nail/home/chloeh/gqltada/node_modules/@gql.tada/internal/dist/gql-tada-internal.js:2227:17)
      at async Object.load (/nail/home/chloeh/gqltada/node_modules/@gql.tada/internal/dist/gql-tada-internal.js:2252:36)
      at async /nail/home/chloeh/gqltada/node_modules/@gql.tada/internal/dist/gql-tada-internal.js:3051:17
      at async Promise.all (index 0)
      at async Object.load (/nail/home/chloeh/gqltada//node_modules/@gql.tada/internal/dist/gql-tada-internal.js:3050:7)
      at async run$1 (/nail/home/chloeh/gqltada/node_modules/@gql.tada/cli-utils/dist/gql-tada-cli.js:4397:5)
      at async convertError (/nail/home/chloeh/gqltada/node_modules/@gql.tada/cli-utils/dist/gql-tada-cli.js:2596:7)
      at async /nail/home/chloeh/gqltada/node_modules/@gql.tada/cli-utils/dist/gql-tada-cli.js:2230:20
```
I have graphql-env.d.ts file generated for me successfully so I assume the path to my schema file is correct but now i get load schema error which is kinda confusing for me. Are there more details we could tell from this error msg? 

FYR here is my tsconfig 

```
{
                "name": "@0no-co/graphqlsp",
                "schema": "../../node_modules/schema.graphql",
                "tadaOutputLocation": "./src/graphql-env.d.ts"
              }
```

**Reply by JoviDeCroock** [reply]
Hmm, you are probably using a version of GraphQL prior to 15.2.0 which would mean that `executeSync` is indeed not present yet. Out of curiosity, what version of GraphQL are you using?

**Reply by kitten** [reply]
That looks like you're missing the peer dependency on `graphql`.
The `@gql.tada/internal` package has a peer dependency on `graphql`, which `@gql.tada/cli-utils` fulfils. I'd recommend you to check the output you got during the installation process, but this seems _very_ unusual.

**Reply by cehan-Chloe** [reply]
yes. I'm using GraphQL 14.7.0. I can upgrade it and try again. Will also check after upgrade if the issues i'm facing also caused by it.

Besides, I retried installation.
```
yarn add gql.tada
yarn add --dev @0no-co/graphqlsp
```
It doesn't give me any warnings or errors.
